src/pingdb/pingdb.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"os"
+	"regexp"
+	"strconv"
+	"time"
+
+	_ "github.com/go-sql-driver/mysql"
+)
+
+func main() {
+	port := os.Getenv("PORT")
+	if port == "" {
+		fmt.Fprintf(os.Stderr, "Error: PORT environment variable is required\n")
+		os.Exit(1)
+	}
+
+	timeoutStr := os.Getenv("TIMEOUT")
```

Comment:
I would just pass a Go duration value here instead of expecting a raw integer.

```go
	if os.Getenv("TIMEOUT") == "" {
		fmt.Fprintln(os.Stderr, "Error: TIMEOUT environment variable is required")
		os.Exit(1)
	}

	var timeout time.Duration
	if v, err := time.ParseDuration(os.Getenv("TIMEOUT")); err == nil {
		timeout = v
	} else {
		fmt.Fprintln(os.Stderr, "Error: invalid TIMEOUT value", os.Getenv("TIMEOUT"))
		os.Exit(1)
	}
...
	ctx, cancel := context.WithTimeout(context.Background(), timeout)
	defer cancel()
...
```

Nit:  Might be worth considering an env configuration parsing library.  I like [github.com/caarlos0/env/v11](https://github.com/caarlos0/env), as it makes the code a little easier to read.

Reply:
✔️